### PR TITLE
Added the ability to copy art to clipboard

### LINKED ISSUE #1: Feature Request: Add a button to copy image to clipboard
So we can click copy, then its in our clipboard and we can shitpost in discord directly.

### PR DESCRIPTION
**Pull request by lambdasapple**
Fixed #1 

TODO: Use a better way to notify the user of the copy rather than alert

### CODE REVIEW

File: src/routes/Generator.svelte [hunk 1]
```diff
@@ -38,24 +38,43 @@
...
 		const canvas = await html2canvas(albumArt);
 
-		let image = await Image.load(canvas.toDataURL('image/jpeg'));
+		let image = await Image.load(canvas.toDataURL('image/png'));
 		const originalWidth = image.width;
 		image = image
 			.resize({ preserveAspectRatio: true, width: originalWidth * 0.85 }) // Get those compression artifacts like the original cover
 			.resize({ preserveAspectRatio: true, width: originalWidth })
-			.blurFilter({ radius: 2 });
+			.blurFilter({ radius: 3 });
```
Comment (Ellipse0934) [inline src/routes/Generator.svelte:50]: @datawater Is this intentional

Comment (datawater) [reply src/routes/Generator.svelte:50]: yes, PNG messes with the blur a bit

File: src/routes/Generator.svelte [hunk 2]
```diff
@@ -38,24 +38,43 @@
...
+		const image = await renderArt();
+		const blob = await image.toBlob('image/png', 1.0);
+
+		const item = new ClipboardItem({
+			'image/png': blob
+		});
+
+		await navigator.clipboard.write([item]);
+
+		alert('Copied image to clipboard');
```
Comment (retronav) [inline src/routes/Generator.svelte:75]: I think if we could show a message below the buttons that'd be great if you want an alternative to this.

File: src/routes/Generator.svelte (hunk 1 above)
Comment (retronav) [inline src/routes/Generator.svelte:50]: I'm not really getting along with this blur radius. 

This is with radius = 2
image

This is with radius = 3
image

The increase in blur is pretty significant for smaller text. For reference, I had checked the deluxe album cover art for the blur radius. I think we could keep the radius as-is.

**Review by retronav — COMMENTED** [review]
Looks good to me overall, the change to PNG is acceptable because the compression artifacts look more or less the same. We can merge this after deciding on the "copied" message.

File: src/routes/Generator.svelte (hunk 1 above)
Comment (Ellipse0934) [reply src/routes/Generator.svelte:50]: How do we settle this conflict ? Maybe set to 2.5 default and make is a variable for the ADVANCED USERS ??

Comment (retronav) [reply src/routes/Generator.svelte]: > How do we settle this conflict ? Maybe set to 2.5 default and make is a variable for the ADVANCED USERS ??

@Ellipse0934 we could indeed have an option to set both the blur radius and the "compressed image" effect but I'm not looking to complicate that and instead keep the result as close to the original album art as possible.

**Comment by retronav** [comment]
@datawater If you still are working on showing a better version for the copy alert here let me know or else this is good to merge.

**Comment by Ellipse0934** [comment]
@datawater PLEASE RESPOND.
LET THE REPO OWNER KNOW THAT YOU ARE DONE. OWO
GOD MORGON btw

**Comment by datawater** [comment]
> @datawater If you still are working on showing a better version for the copy alert here let me know or else this is good to merge.

We could use something like [sweetalert](https://sweetalert2.github.io/), but I feel like it would bloat the webpage too much (it's about 55KBs). I pass the decision to you, if you want me to implement a better way, LMK, else you can merge it.

**Comment by retronav** [comment]
@datawater I'll add a better implementation for the message if I find any. For the time being,

merging that

**Pull request merged**